packages/block-library/src/navigation-menu-item/edit.js

```diff
@@ -29,71 +29,124 @@ import {
-} from '@wordpress/element';
+import { Fragment, useState, useRef, useEffect } from '@wordpress/element';
 
 function NavigationMenuItemEdit( {
 	attributes,
 	isSelected,
 	isParentOfSelectedBlock,
 	setAttributes,
+	fetchSearchSuggestions,
 } ) {
-	const plainTextRef = useRef( null );
+	const { label, opensInNewTab, title, url } = attributes;
+	const link = title ? { title, url } : null;
 	const [ isLinkOpen, setIsLinkOpen ] = useState( false );
-	const [ isEditingLink, setIsEditingLink ] = useState( false );
-	const [ urlInput, setUrlInput ] = useState( null );
+	const [ wasCloseByLinkControl, setWasCloseByLinkControl ] = useState( false );
+
+	const plainTextRef = useRef( null );
+
+	/**
+	 * It's a kind of hack to handle closing the LinkControl popover
+	 * clicking on the ToolbarButton link.
+	 */
+	useEffect( () => {
+		if ( ! isSelected ) {
+			setIsLinkOpen( false );
+			setWasCloseByLinkControl( false );
+		}
+		return () => {
+			setIsLinkOpen( false );
+			setWasCloseByLinkControl( false );
+		};
+	}, [ isSelected ] );
 
-	const inputValue = urlInput !== null ? urlInput : url;
+	/**
+	 * `onKeyDown` LinkControl handler.
+	 * It takes over to stop the event propagation to make the
+	 * navigation work, avoiding undesired behaviors.
+	 * For instance, it will block to move between menu items
+	 * when the LinkOver is focused.
+	 *
+	 * @param {Event} event
+	 */
+	const handleLinkControlOnKeyDown = ( event ) => {
+		const { keyCode } = event;
 
-	const onKeyDown = ( event ) => {
-		if ( [ LEFT, DOWN, RIGHT, UP, BACKSPACE, ENTER ].indexOf( event.keyCode ) > -1 ) {
+		if ( [ LEFT, DOWN, RIGHT, UP, BACKSPACE, ENTER ].indexOf( keyCode ) > -1 ) {
 			// Stop the key event from propagating up to ObserveTyping.startTypingInTextField.
 			event.stopPropagation();
 		}
 	};
 
-	const closeURLPopover = () => {
-		setIsEditingLink( false );
-		setUrlInput( null );
-		setIsLinkOpen( false );
-	};
-
-	const autocompleteRef = useRef( null );
-
-	const onFocusOutside = ( event ) => {
-		const autocompleteElement = autocompleteRef.current;
-		if ( autocompleteElement && autocompleteElement.contains( event.target ) ) {
-			return;
+	/**
+	 * Updates the link attribute when it changes
+	 * through of the `onLinkChange` LinkControl callback.
+	 *
+	 * @param {Object|null} itemLink Link object if it has been selected. Otherwise, Null.
+	 */
+	const updateLink = ( itemLink ) => {
+		if ( ! itemLink ) {
+			// If not link, then set empty string to link attrs.
+			return setAttributes( { title: '', url: '' } );
 		}
-		closeURLPopover();
+
+		setAttributes( { title: itemLink.title, url: itemLink.url } );
 	};
 
-	const stopPropagation = ( event ) => {
-		event.stopPropagation();
+	/**
+	 * It updates the link attribute when the
+	 * link settings changes.
+	 *
+	 * @param {string} setting Setting type, for instance, `new-tab`.
+	 * @param {string} value Setting type value.
+	 */
+	const updateLinkSetting = ( setting, value ) => {
+		if ( 'new-tab' ) {
```

Comment:
Seems as though this will always be `truthy`. New tab is the only setting, so I guess it hasn't caused an issue.

```suggestion
		if ( setting === 'new-tab' ) {
```

Reply:
I think `LinkControl` needs a refactor in light of this. Nothing major right now but we need to make these controls extensible.

Also @talldan is right this will always be truthy 😄

Reply:
Fixed